activiti-spring-process-extensions/src/main/java/org/activiti/spring/process/variable/DateFormatterProvider.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.activiti.spring.process.variable;
+
+import java.text.MessageFormat;
+import java.time.*;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.util.Date;
+
+public class DateFormatterProvider  {
+   
+    private String dateFormatPattern;
+    
+    private ZoneId zoneId = ZoneOffset.UTC;
+ 
+    public DateFormatterProvider(String dateFormatPattern) {
+        this.dateFormatPattern = dateFormatPattern;
+    }
+
+    public ZoneId getZoneId() {
+        return zoneId;
+    }
+    
+    public String getDateFormatPattern() {
+        return dateFormatPattern;
+    }
+    
+    public void setDateFormatPattern(String dateFormatPattern) {
+        this.dateFormatPattern = dateFormatPattern;
+    }
+ 
+    public Date parse(String value) throws DateTimeException {
+        DateTimeFormatter dateTimeFormatter = new DateTimeFormatterBuilder()
+                                                      .appendPattern(getDateFormatPattern())
+                                                      .toFormatter()
+                                                      .withZone(getZoneId());
+        
+        try {
+            LocalDateTime localDateTime = dateTimeFormatter.parse(value,
+                                                                  LocalDateTime::from);
+            return Date.from(localDateTime.atZone(getZoneId()).toInstant());                
+        } catch (DateTimeException e) {
```

Comment:
will catch input values like "1970-01-01" ? 
I was just wondering if it is necessary.. i was expecting localdatetime to return "1970-01-01T00:00:00.000Z" for an input like "1970-01-01"

Reply:
It is possible that input string will come in the date format (without time, for example from extension file): "1970-01-01"... And we have to convert this kind of strings to Date...

Reply:
"1970-01-01" gets converted to "1970-01-01T00:00:00.000Z" so i think they should be the same thing, because if i am not wrong they both should have => hasTime(0)

We can leave it if you want, or we can add a test to verify that having only LocalDateTime does not break things. Up to you.

Reply:
Good point, @ffazzini. The test is covering this case, so we can remove the catch block and see if the test is still passing. If that's the case we can remove the catch block.

Reply:
>"1970-01-01" gets converted to "1970-01-01T00:00:00.000Z" so i think they should be the same thing, because if i am not wrong they both should have => hasTime(0)

Yes, true, they will both have hasTime(0)...

But, parsing of this string "1970-01-01" using a code below throw exception:
```
LocalDateTime localDateTime = dateTimeFormatter.parse(value,
                                                                  LocalDateTime::from);
            return Date.from(localDateTime.atZone(getZoneId()).toInstant()); 
```
That's why we are using this part:
```
} catch (DateTimeException e) {
            LocalDate localDate = dateTimeFormatter.parse(String.valueOf(value),
                                                          LocalDate::from);
            return Date.from(localDate.atStartOfDay().atZone(getZoneId()).toInstant());
        }
```

Reply:
Strings like "1970-01-01" this may come as variable value from any place, and we should handle this, I think...

Reply:
yes, we should keep it in this case.

Reply:
ok